pkg/ccl/streamingccl/streamingest/stream_ingestion_dist_test.go

```diff
@@ -316,3 +316,139 @@ func TestSourceDestMatching(t *testing.T) {
+		name            string
+		partitionSpecs  execinfrapb.StreamIngestionPartitionSpecs
+		frontierEntries execinfrapb.FrontierEntries
+		expected        []frontierExecutionDetails
+	}{
+		{
+			name: "matching spans",
+			partitionSpecs: execinfrapb.StreamIngestionPartitionSpecs{
+				Specs: []*execinfrapb.StreamIngestionPartitionSpec{
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 2,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+						},
+					},
+				}},
+			frontierEntries: execinfrapb.FrontierEntries{ResolvedSpans: []jobspb.ResolvedSpan{
+				{
+					Span:      roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+					Timestamp: hlc.Timestamp{WallTime: 1},
+				},
+			}},
+			expected: []frontierExecutionDetails{
+				{
+					srcInstanceID:  1,
+					destInstanceID: 2,
+					span:           "{a-b}",
+				}},
+		},
+		{
+			name: "multi-partition",
+			partitionSpecs: execinfrapb.StreamIngestionPartitionSpecs{
+				Specs: []*execinfrapb.StreamIngestionPartitionSpec{
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 2,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+						},
+					},
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 3,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("b"), EndKey: roachpb.Key("c")},
+						},
+					},
+				}},
+			frontierEntries: execinfrapb.FrontierEntries{ResolvedSpans: []jobspb.ResolvedSpan{
+				{
+					Span:      roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a'")},
+					Timestamp: hlc.Timestamp{WallTime: 1},
+				},
+				{
+					Span:      roachpb.Span{Key: roachpb.Key("b"), EndKey: roachpb.Key("c")},
+					Timestamp: hlc.Timestamp{WallTime: 2},
+				},
+			}},
+			expected: []frontierExecutionDetails{
+				{
+					srcInstanceID:  1,
+					destInstanceID: 2,
+					span:           "a{-'}",
+				},
+				{
+					srcInstanceID:  1,
+					destInstanceID: 3,
+					span:           "{b-c}",
+				},
+			},
+		},
+		{
+			name: "merged frontier",
+			partitionSpecs: execinfrapb.StreamIngestionPartitionSpecs{
+				Specs: []*execinfrapb.StreamIngestionPartitionSpec{
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 2,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+						},
+					},
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 2,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("b"), EndKey: roachpb.Key("d")},
+						},
+					},
+				}},
+			frontierEntries: execinfrapb.FrontierEntries{ResolvedSpans: []jobspb.ResolvedSpan{
+				{
+					Span:      roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("d")},
+					Timestamp: hlc.Timestamp{WallTime: 1},
+				},
+			}},
+			expected: []frontierExecutionDetails{
+				{
+					srcInstanceID:  1,
+					destInstanceID: 2,
+					span:           "{a-b}",
+				},
+				{
+					srcInstanceID:  1,
+					destInstanceID: 2,
+					span:           "{b-d}",
+				}},
+		},
```

Comment:
could you add a test case where a partitionSpec's Span has a split frontier? i.e.
```
partitionSpecs: execinfrapb.StreamIngestionPartitionSpecs{
				Specs: []*execinfrapb.StreamIngestionPartitionSpec{
					{
						SrcInstanceID:  1,
						DestInstanceID: 2,
						Spans: []roachpb.Span{
							{Key: roachpb.Key("a"), EndKey: roachpb.Key("c")},
						},
					},
				}},
			frontierEntries: execinfrapb.FrontierEntries{ResolvedSpans: []jobspb.ResolvedSpan{
				{
					Span:      roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
					Timestamp: hlc.Timestamp{WallTime: 1},
				},
{
					Span:      roachpb.Span{Key: roachpb.Key("b"), EndKey: roachpb.Key("c")},
					Timestamp: hlc.Timestamp{WallTime: 2},
				},
			}},
```